src/com/palmergames/bukkit/towny/tasks/DrawSmokeTask.java

```diff
@@ -11,9 +20,13 @@
 import com.palmergames.bukkit.towny.object.WorldCoord;
 import com.palmergames.bukkit.towny.utils.BorderUtil;
 import com.palmergames.bukkit.util.DrawSmokeTaskFactory;
+import com.palmergames.util.TimeMgmt;
 
 public class DrawSmokeTask extends TownyTimerTask {
 
+	Map<UUID, List<CellBorder>> townCellBorderMap = new HashMap<>();
+	Map<UUID, Long> townCachedTime = new HashMap<>();

```

Comment:
This currently causes a small memory leak since cellborders are never removed from the map, a guava cache might be better.